framework/src/araui/core/ui_core.c

```diff
@@ -211,51 +212,61 @@ static ui_error_t _ui_process_widget_recur(ui_widget_body_t *widget, uint32_t dt
 		return UI_INVALID_PARAM;
 	}
 
-	if (widget->tween_cb) {
-		if (widget->tween_info.t > widget->tween_info.d) {
-			widget->tween_info.t = widget->tween_info.d;
-			widget->tween_cb((ui_widget_t)widget, widget->tween_info.t);
-			widget->tween_cb = UI_NULL;
+	ui_widget_queue_init();
+	ui_widget_queue_push(widget);
 
-			if (ui_request_callback(_ui_call_tween_finished_cb, widget) != UI_OK) {
-				UI_LOGE("Error: cannot make a request to call.tween_finished_cb!\n");
+	while (!ui_widget_is_queue_empty()) {
```

Comment:
Does stack push none-widget component? If it works like that, it's ok but if it always push widget, then it is error case if popped object is not type of widget.
In this case you have to print out error message and stop this.

Reply:
As the prototype of `ui_widget_queue_dequeue()` is `ui_widget_body_t *ui_widget_queue_dequeue(void)`, the return value of `ui_widget_queue_dequeue()` is always in the type of widget.

Reply:
I meaning below code, curr_widget can be null in queue or stack event it is not empty?
```
		curr_widget = ui_widget_queue_dequeue();
		if (!curr_widget) {
			continue;
		}
```